src/renderers/shared/fiber/ReactFiberBeginWork.js

```diff
@@ -464,13 +466,21 @@ module.exports = function<T, P, I, TI, C, CX, CI>(
             info += ' Check the render method of `' + ownerName + '`.';
           }
 
-          warning(
-            false,
-            'Stateless function components cannot be given refs. ' +
-            'Attempts to access this ref will fail.%s%s',
-            info,
-            ReactDebugCurrentFiber.getCurrentFiberStackAddendum()
-          );
+          let warningKey = ownerName || workInProgress._debugID || '';
+          const debugSource = workInProgress._debugSource;
```

Comment:
It is added [here](https://github.com/facebook/react/blob/280bcfa9f596a6747bdb99c5e416e924bc400644/src/renderers/shared/fiber/ReactFiber.js#L311) to the Fiber, and it is on the element from [this Babel plugin](https://www.npmjs.com/package/babel-plugin-transform-react-jsx-source) (enabled by default in CRA).